datafusion/expr/src/udf.rs

```diff
@@ -450,6 +451,7 @@ pub trait ScalarUDFImpl: Debug + Send + Sync {
     /// is recommended to return [`DataFusionError::Internal`].
     ///
     /// [`DataFusionError::Internal`]: datafusion_common::DataFusionError::Internal
+    #[deprecated(since = "44.0.0", note = "Use `return_type_from_exprs` instead")]
```

Comment:
if we are going to deprecate this API I think we should add an example to `return_type_from_exprs` to help the migration effort.

I can help if we proceed